src/scripts/main.js

```diff
@@ -1 +1,106 @@
 'use strict';
+
+const bodyDoc = document.querySelector('body');
+let rightClick;
+let leftClick;
+
+function makeDiv(className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.setAttribute('class', className);
+  bodyDoc.appendChild(div);
+
+  return div;
+}
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.addEventListener(
+      'click',
+      () => {
+        leftClick = true;
+
+        resolve('First promise was resolved');
+      },
+      { once: true },
+    );
+
+    // eslint-disable-next-line prefer-promise-reject-errors
+    setTimeout(() => reject('First promise was rejected!'), 3000);
+  });
+};
+
+const secondPromise = () => {
+  return new Promise((resolve) => {
+    document.addEventListener(
+      'click',
+      () => {
+        leftClick = true;
+
+        resolve('Second promise was resolved');
+      },
+      { once: true },
+    );
+
+    document.addEventListener(
+      'contextmenu',
+      () => {
+        rightClick = true;
+
+        resolve('Second promise was resolved');
+      },
+      { once: true },
+    );
+  });
+};
+
+const thirdPromise = () => {
+  return new Promise((resolve) => {
+    document.addEventListener(
+      'click',
+      () => {
+        if (rightClick === true) {
+          rightClick = false;
+          resolve('Third promise was resolved');
+        }
+      },
+      { once: true },
+    );
+
+    document.addEventListener(
+      'contextmenu',
+      () => {
+        if (leftClick === true) {
+          leftClick = false;
+          resolve('Third promise was resolved');
```

Comment:
The `leftClick` variable is checked, but since it's never set to `true` within the `firstPromise` function, this condition will never be met if `thirdPromise` is called immediately after `firstPromise`. Ensure that the logic correctly reflects the intended behavior.